controllers/awsmachine_controller.go

```diff
@@ -373,6 +379,40 @@ func (r *AWSMachineReconciler) getOrCreate(scope *scope.MachineScope, ec2svc ser
 	return instance, nil
 }
 
+func (r *AWSMachineReconciler) getAddresses(instance *infrav1.Instance, ec2svc *ec2.Service) ([]corev1.NodeAddress, error) {
```

Comment:
I wonder if this should be part of the describe instance rather than having the AWS-specific logic in the reconciler